### PR TITLE
fix(all-dashboard): create combined distinct-value-handler

### PR DESCRIPTION
**Pull request by piggggggggy**
### To Reviewers
- [ ] Skip (`style`, `chore` ONLY)
- [ ] Not that difficult

### Type of Change
- [ ] New feature
- [x] Bug fixes
- [x] Feature improvement
- [ ] Refactor
- [ ] Others (performance improvement, refactoring, CI/CD, etc.)

### Checklist
- [x] `Error / Warning / Lint / Type`

### Description
SSIA

[QA - 320 - 1](https://pyengine.atlassian.net/browse/QA-320)

- Create `combinedDashbaordLabelsAutoCompleteHandler` and apply.

### Things to Talk About

There are no specific sorting criteria. However, since there is no priority between project and domain, the most basic ascending order was applied.

### CODE REVIEW

File: src/services/dashboards/all-dashboards/AllDashboardsPage.vue [hunk 1]
```diff
@@ -170,6 +172,39 @@ export default {
             });
         };
 
+        const combinedDashbaordLabelsAutoCompleteHandler = (): ValueHandler | undefined => {
+            const projectLabelsValueHandler = makeDistinctValueHandler('dashboard.ProjectDashboard', 'labels');
+            const domainLabelsValueHandler = makeDistinctValueHandler('dashboard.DomainDashboard', 'labels');
```
Comment (sulmoJ) [inline src/services/dashboards/all-dashboards/AllDashboardsPage.vue:177]: Suggestion: This handler seems to require filters for `Viewer` or `Scope` options.

**Review by sulmoJ — COMMENTED** [review]
I left a comment!

**Review by sulmoJ — APPROVED** [review]
LGTM!!
Thank you for the detailed explanation!

File: src/services/dashboards/all-dashboards/AllDashboardsPage.vue [hunk 2]
```diff
@@ -170,6 +172,39 @@ export default {
...
+            if (!projectLabelsValueHandler && !domainLabelsValueHandler) return undefined;
+
+            return async (inputText: string, keyItem: KeyItem, currentDataType?: KeyDataType, subPath?: string) => {
+                const defaultResult = {
+                    results: [],
+                    totalCount: 0,
+                };
+                const projectResult = projectLabelsValueHandler ? await projectLabelsValueHandler(inputText, keyItem, currentDataType, subPath) : defaultResult;
+                const domainResult = domainLabelsValueHandler ? await domainLabelsValueHandler(inputText, keyItem, currentDataType, subPath) : defaultResult;
```
Comment (WANZARGEN) [inline src/services/dashboards/all-dashboards/AllDashboardsPage.vue:186]: suggestion: I suggest to use promise.allSettled() to await promise in parallel.

```suggestion
const promises = [];
if (projectLabelsValueHandler instanceof Promise) promises.push(projectLabelsValueHandler)
else ...
if (domainLabelsValueHandler instanceof Promise) promises.push(domainLabelsValueHandler)
else ...
const results = await Promise.allSettled(promises);
results.map(result => { ... })
```

File: src/services/dashboards/all-dashboards/AllDashboardsPage.vue [hunk 3]
```diff
@@ -170,6 +172,39 @@ export default {
...
+                    results = [...results, result];
+                });
+                results.sort((a, b) => {
+                    if (a.label > b.label) return 1;
+                    return -1;
+                });
+
+                return {
+                    results,
+                    totalCount: (projectResult.totalCount ?? 0) + (domainResult.totalCount ?? 0),
```
Comment (WANZARGEN) [inline src/services/dashboards/all-dashboards/AllDashboardsPage.vue:203]: suggestion: totalCount is deprecated feature. Please use 'more' :)

**Review by WANZARGEN — APPROVED** [review]
I left some suggestions!!

**Comment by piggggggggy** [comment]
Applied reiviews. Thank you for reviewing!!

**Pull request merged**